make/env.mk

```diff
@@ -40,10 +40,14 @@ endif
 TAG := # make sure tag is never injectable as an env var
 RELEASE_GOTAGS := release
 
-# Use a release go -tag when CI is targetting a tag
+# Use a release go -tag when CI is targeting a tag
 ifdef CI
 ifneq ($(BUILD_TAG),)
+ifeq ($(GOTAGS),)
 GOTAGS := $(RELEASE_GOTAGS)
+else
+GOTAGS := $(GOTAGS),$(RELEASE_GOTAGS)
```

Comment:
This looks clumsy. Is there a reason why we would _not_ want `strictfipsruntime` to be always on for any builds?

If no reason, it's perhaps simpler to add both the env variable and a tag directly in
https://github.com/stackrox/stackrox/blob/784f3d253f7194f2e1bce5767c9ee8d5f58fe464/scripts/go-tool.sh#L64-L72

Reply:
> Is there a reason why we would not want strictfipsruntime to be always on for any builds?

I think if we add it to the release defaults, that would be ideal.

It seems extremely unlikely, but I'm wondering if there's a case where if a developer has a go compiler that recognizes this tag, it won't link properly (my current thought is due to not finding the appropriate openssl library). I'm still fuzzy on those details so I need to dig in further if that's a valid concern. If we only add it to the release defaults, maybe we don't need to worry about it. 

> This looks clumsy.
I agree; I'm not the biggest fan of this.

It was pretty jarring to set `GOTAGS` and have it be overwritten, though, so I'd like to figure out a way to either make the behavior of the `GOTAGS` variable more intuitive or somehow inform the user that setting it with the `CI` and non-empty `BUILD_TAG` won't do anything (maybe a simple log would suffice?).

Reply:
> It seems extremely unlikely, but I'm wondering if there's a case where if a developer has a go compiler that recognizes this tag, it won't link properly (my current thought is due to not finding the appropriate openssl library). I'm still fuzzy on those details so I need to dig in further if that's a valid concern. If we only add it to the release defaults, maybe we don't need to worry about it.

Sorry, this reads too uncertain. My concern is that at this point we're selecting among the solutions that look safe without knowing is there a problem at all.

We have the following dimensions:
* `strictfipsruntime`: set or not.
* Go compiler: FIPS-pached openshift-golang-builder or else.
* Runtime cluster: FIPS-enabled or not.
* CGO: enabled or disabled (the latter if for downloadable `roxctl`-s).

If some combination causes problems, we should certainly know about that before handing the results to customers and before declaring ROX-24283 as done.
I'm not saying testing the full matrix is part of this PR but I'm more for taking the least cautious route (which is also more straightforward code) and causing breakages earlier so that we become better informed. And certainly diving deeper in the mechanics of all this.